### PR TITLE
fix: parse error due incrorrect csv response from marketo-n

### PR DESCRIPTION
**Pull request by utsabc**
# Description

The issue is due to incorrect csv response from marketo, hence rather than reading all rows at once we are reading each row separately. 

For the erroneous rows we cannot identify which particular job as we cant generate hash to get corresponding job id hence these specific ones will be marked as success (we need to figure out a long term solution for this though)
Screenshot 2024-10-30 at 7 19 48 PM

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go [hunk 1]
```diff
@@ -208,9 +208,19 @@ func (m *MarketoAPIService) GetLeadStatus(url string) ([]map[string]string, *API
...
-	if err != nil {
-		return nil, &APIError{StatusCode: 500, Category: "Retryable", Message: "Error in parsing csv response"}
+	// read each row one by one
+	rows := make([][]string, 0)
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
```
Comment (koladilip) [inline router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go:218]: looks even error comes, record is returned, so can use it, can you try this by sample code

Comment (koladilip) [reply router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go:218]: we should handle this by removing \n characters in transformer

**Review by koladilip — APPROVED** [review]

File: router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go (hunk 1 above)
Comment (utsabc) [reply router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go:218]: Sure, will take it up

File: router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go [hunk 2]
```diff
@@ -208,9 +208,19 @@ func (m *MarketoAPIService) GetLeadStatus(url string) ([]map[string]string, *API
...
+			m.logger.Error("Error in parsing csv response", err.Error())
```
Comment (krishna2020) [inline router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/apiService.go:219]: Is line number captured in err.Error() ?

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**